overlay/default.nix

```diff
@@ -12,6 +12,48 @@ in
   # Uses sources.mix-to-nix and sources.gitignore
   nixUnstable = inputs.nix.defaultPackage.${final.system};
 
+  benchwrapper = prev.writeShellScriptBin "benchwrapper" ''
+    set -euo pipefail
+
+    # number of cores to run the wrapped program on
+    test_cpus=2
```

Comment:
Let's make this configurable with some sane checks that one doesn't try to allocate more than some amount of CPUs (let's say not more than half of the overall amount)

Reply:
I was thinking about the best way to solve this. we could make this a function instead that takes a numerical argument for the number of cores

the check would be necessary then

Reply:
I'm not sure if an additional argument will be convenient for users, I was thinking about env variable instead

Reply:
env variable, of course :facepalm: 

did not think of that :eyes:

Reply:
env variables don't work as well with nix though, unless in impure mode :thinking:

Reply:
Oh, indeed, then a dedicated argument is the way

Reply:
I wonder how that might affect usability, entering the number of cores you request every time :thinking: 
considering it's our only option really and that this wrapper is intended for longer running tasks it should be fine I think

Reply:
I believe it shouldn't be very inconvenient as long as there is an instruction that explains how to use this script with some usage examples